Documentation/concepts/networking/masquerading.rst

```diff
@@ -32,29 +32,111 @@ Setting the routable CIDR
 
 Implementation Modes
 --------------------
 
 eBPF-based
-  The eBPF-based implementation is the most efficient
-  implementation. It requires Linux kernel 4.19 and can be enabled with the
-  option ``enable-bpf-masquerade: true``.
+**********
+
+The eBPF-based implementation is the most efficient
+implementation. It requires Linux kernel 4.19 and can be enabled with
+the ``global.bpfMasquerade=true`` helm option (enabled by default).
+
+The current implementation depends on :ref:`the BPF NodePort feature <kubeproxy-free>`.
+The dependency will be removed in the Cilium v1.9 release.
+
+Masquerading can take place only on those devices which run the eBPF masquerading
+program. This means that a packet sent from a pod to an outside will be masqueraded
+(to an output device IPv4 address), if the output device runs the program. If not
+specified, the program will be automatically attached to the devices selected by
+:ref:`the BPF NodePort device detection metchanism <Nodeport Devices>`.
+To manually change this, use the ``global.devices`` helm option. Use ``cilium status``
+to determine which devices the program is running on:
+
+::
+
+    kubectl exec -it -n kube-system cilium-xxxxx -- cilium status | grep Masquerading
+    Masquerading:   BPF (ip-masq-agent)   [eth0, eth1]  10.0.0.0/16
+
+From the output above, the program is running on the ``eth0`` and ``eth1`` devices.
+
+
+The eBPF-based masquerading can masquerade packets of the following IPv4 L4 protocols:
+
+- TCP
+- UDP
+- ICMP (only Echo request and Echo reply)
+
+By default, any packet from a pod destined to an IP address outside of the
+``native-routing-cidr`` range is masqueraded. The exclusion CIDR is shown in the above
+output of ``cilium status`` (``10.0.0.0.16``).  To allow more fine-grained control,
+Cilium implements `ip-masq-agent <https://github.com/kubernetes-sigs/ip-masq-agent>`_
+in eBPF which can be enabled with the ``global.ipMasqAgent.enabled=true`` helm option.
+
+The eBPF-based ip-masq-agent supports the ``nonMasqueradeCIDRs`` and
+``masqLinkLocal`` options set in a configuration file. A packet sent from a pod to
+a destination which belongs to any CIDR from the ``nonMasqueradeCIDRs`` is not
+going to be masqueraded. If the configuration file is empty, the agent will provision
+the following non-masquerade CIDRs:
+
+- ``10.0.0.0/8``
+- ``172.16.0.0/12``
+- ``192.168.0.0/16``
+- ``100.64.0.0/10``
+- ``192.0.0.0/24``
+- ``192.0.2.0/24``
+- ``192.88.99.0/24``
+- ``198.18.0.0/15``
+- ``198.51.100.0/24``
+- ``203.0.113.0/24``
+- ``240.0.0.0/4``
+
+In addition, if the ``masqLinkLocal`` is not set or set to false, then
+``169.254.0.0/16`` is appended to the non-masquerade CIDRs list.
+
+The agent uses Fsnotify to track updates to the configuration file, so the original
+``resyncInterval`` option is unnecessary.
+
+I think it's important for users to understand how we sync a config. The example below
+shows how to configure the agent via `ConfigMap` and to verify it:
+
+::
```

Comment:
Any reason to avoid using `.. code-block:: shell-session` and obvious shell bits like starting lines you execute with `$`? Seems a bit weird to not differentiate the commands and the output here.

Reply:
Agree, but I wanted to make it consistent with other guides. IIRC we had a discussion some time ago whether we should keep `$`'s in our guides. The consensus was to get rid of them.